### ISSUE TITLE
SharePoint 2019 On Premise ClientsidePageFromFile bring error

### ISSUE THREAD

**Issue by tntboneman** [pnp/pnpjs#1349]
### Category
- [ ] Enhancement
- [X] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

pnpjs:2.0.7
spfx/@microsoft/sp:1.4.1
sharepoint server 2019 on premise
typescript:3.6.4

### Expected / Desired Behavior / Question
I want to use pnpjs copying pages form template pages but when I new 'ClientsidePageFromFile' object, it returns 'Cannot read property 'zoneEmphasis' of undefined'
I saw the article(https://github.com/pnp/pnpjs/issues/1111), it says this bug was solved in 2.0.4 but I tried 1.3.X - 2.0.7 this feature all fails and return the same error.

### Observed Behavior
```js
import { sp } from '@pnp/sp';
import '@pnp/sp/webs';
import { ClientsideText, ClientsideWebpart, ClientsidePageFromFile } from '@pnp/sp/clientside-pages';
import '@pnp/sp/comments/clientside-page';

...
const templatePage = await ClientsidePageFromFile(sp.web.getFileByServerRelativeUrl(templatePageUrl));//error here, sp.web.getFileByServerRelativeUrl(templatePageUrl) has value
await templatePage.copy(sp.web, pagename + ".aspx", pagename, false);
...
```

**Comment by juliemturner** [comment]
Based on the screen shot that you provided in the other issue (thanks btw for starting a new issue and referencing it as we do not look at new comments on closed issues), I believe that you uncovered another location in the code that needs to check if the emphasis object was undefined. We do not have a way to test on premises server so we are somewhat coding blind but we'll include the update in the next release and if you continue to have issues you can submit a new bug, or better yet, you could test it yourself and submit a PR that you know is completely working!! 😀

**Comment by tntboneman** [comment]
Thank you, juliemturner.

After test, my situation was caused by this reason:
* >the template page which I want to copy has a empty column(perhaps the page has a empty section and not set any webparts), in clientsidePage logic it won't has the structure"xxx.emphasis.zoneEmphasis".

So I add a logic in @pnp/sp/clientside-pages/types.js line 680, then it works.
```js
switch (controlType) {
    case 0:
        // empty canvas column or page settings
        if (hOP(controls[i], "pageSettingsSlice")) {
            this._pageSettings = controls[i];
        }
        else {
            // we have an empty column
            //line 680 add this logic, copy file will not throw error
            if (!controls[i].emphasis) {
                controls[i].emphasis = { zoneEmphasis: 0 };
            }
            this.mergeColumnToTree(new CanvasColumn(controls[i]));
        }
        break;
        ...
}
```
BTW, the pnpjs version 2.0.8 cannot support @microsoft/sp 1.4.1, it will throw a error "Property 'msGraphClientFactory' is missing in type 'ListView Command SetContext' but required in type 'ISPFXContext'" but sharepoint server 2019 on premise only support 1.4.1, so I can just use pnpjs 2.0.7.

**Comment by juliemturner** [comment]
So, yes, the fix you manually put in will be in the 2.0.9 version... you should be able to use that when it releases on Friday.

Regarding your comment about the msGraphClientFactory, you may have an issue there, but I don't think it'll be because of the version of the library (2.0.7 vs 2.0.8).  Ideally you would open a new issue and share code that explains the situation in more detail.

**Comment by juliemturner** [comment]
Ok, I see.  That property was added in 2.0.8 and is not marked optional.  @patrick-rodgers has fixed that in PR #1352 which will be part of 2.0.9 release, which will go out on Friday.

**Comment by tntboneman** [comment]
Thank you juliemturner, I fully expect 2.0.9 version.
